src/Component/src/Metadata/Metadata.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata;
+
+use Doctrine\Inflector\Inflector as InflectorObject;
+use Doctrine\Inflector\InflectorFactory;
+
+final class Metadata implements MetadataInterface
+{
+    private string $name;
+
+    private string $applicationName;
+
+    /** @var string */
+    private $driver;
+
+    private ?string $stateMachineComponent = null;
```

Comment:
I'm not sure about this, but since this property being initialized in the constructor, default value is not necessary.

```suggestion
    private ?string $stateMachineComponent;
```